src/logger.ts

```diff
@@ -88,22 +91,35 @@ export class Logger {
     for (let i = 0; i < options.levels.length; i++) {
       const level = options.levels[i];
       if (i <= levelIndex) {
-        this[level] = (...args) => {
+        // ts: this doesn't have an index signature, but we want to set
+        // properties anyway.
+        // tslint:disable-next-line:no-any
+        (this as any)[level] = (...args: any[]) => {
           args.unshift(level);
           console.log(this[kFormat].apply(this, args));
           return this;
         };
       } else {
-        this[level] = () => this;
+        // tslint:disable-next-line:no-any
+        (this as any)[level] = () => this;
       }
     }
   }
 
-  // TODO: Mark this as protected when TypeScript 2.9 comes out.
   // tslint:disable-next-line:no-any
-  [kFormat](level: string, ...args: any[]): string {
+  private[kFormat](level: string, ...args: any[]): string {
```

Comment:
Nit. A space between `private` and `[kFormat]`. Or is this what clang-format does?

Reply:
This is what clang-format does.